### PR TITLE
8263260: [s390] Support latest hardware (z14 and z15)

### PR DESCRIPTION
**Pull request by RealLucy**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8263260](https://bugs.openjdk.java.net/browse/JDK-8263260): [s390] Support latest hardware (z14 and z15)

### Reviewers
 * [Goetz Lindenmaier](https://openjdk.java.net/census#goetz) (@GoeLin - **Reviewer**) ⚠️ Review applies to c8dcfb447d87669e9e77abdead6ec9997474e958
 * [Martin Doerr](https://openjdk.java.net/census#mdoerr) (@TheRealMDoerr - **Reviewer**)

### Download
To checkout this PR locally:
`$ git fetch https://git.openjdk.java.net/jdk pull/2918/head:pull/2918`
`$ git checkout pull/2918`

To update a local copy of the PR:
`$ git checkout pull/2918`
`$ git pull https://git.openjdk.java.net/jdk pull/2918/head`

### CODE REVIEW

**Comment by RealLucy** [comment]
Dear Community,
I would appreciate reviews for this enhancement, adding basic detection and support for recent s390 hardware generations. 
Thank you!
Lutz

**Review by GoeLin — APPROVED** [review]
LGTM

File: src/hotspot/cpu/s390/vm_version_s390.cpp [hunk 1]
```diff
@@ -48,9 +47,11 @@ unsigned int  VM_Version::_nmsgdigest_features                      = 0;
...
 
-static const char* z_gen[]     = {"  ",   "G1",   "G2", "G3",    "G4",     "G5",      "G6",   "G7"   };
-static const char* z_machine[] = {"  ", "2064", "2084", "2094",  "2097",   "2817",    "  ",   "2964" };
-static const char* z_name[]    = {"  ", "z900", "z990", "z9 EC", "z10 EC", "z196 EC", "ec12", "z13"  };
+static const char* z_gen[]     = {"  ", "G1",         "G2",         "G3",         "G4",         "G5",         "G6",         "G7",         "G8",         "G9"  };
+static const char* z_machine[] = {"  ", "2064",       "2084",       "2094",       "2097",       "2817",       "2827",       "2964",       "3906",       "8561" };
+static const char* z_name[]    = {"  ", "z900",       "z990",       "z9 EC",      "z10 EC",     "z196 EC",    "ec12",       "z13",        "z14",        "z15" };
+static const char* z_WDFM[]    = {"  ", "2006-06-30", "2008-06-30", "2010-06-30", "2012-06-30", "2014-06-30", "2016-12-31", "2019-06-30", "2021-06-30", "tbd" };
+static const char* z_EOS[]     = {"  ", "2014-12-31", "2014-12-31", "2017-10-31", "2019-12-31", "2021-12-31", "tbd",        "tbd",        "tbd",        "tbd" };
```
Comment (TheRealMDoerr) [inline src/hotspot/cpu/s390/vm_version_s390.cpp:54]: Table provides a nice overview, but seems like only z_name is used in the code. The rest only serves as comments.

File: src/hotspot/cpu/s390/vm_version_s390.cpp [hunk 2]
```diff
@@ -285,27 +296,27 @@ void VM_Version::set_features_string() {
...
   }
   if (is_z10()) {
-    _features_string = "system-z, g4-z10, ldisp_fast, extimm, pcrel_load/store, cmpb";
+    _features_string = "system-z, g4-z10, ldisp_fast, extimm, pcrel_load/store, cmpb, out-of-support_as_of_2019-12-31";
     _model_string = z_name[4];
     ambiguity++;
   }
   if (is_z9()) {
-    _features_string = "system-z, g3-z9, ldisp_fast, extimm, out-of-support as of 2016-04-01";
+    _features_string = "system-z, g3-z9, ldisp_fast, extimm, out-of-support_as_of_2016-04-01";
```
Comment (TheRealMDoerr) [inline src/hotspot/cpu/s390/vm_version_s390.cpp:309]: How does this relate to the table above?

File: src/hotspot/cpu/s390/vm_version_s390.hpp [hunk 3]
```diff
@@ -44,6 +44,8 @@ class VM_Version: public Abstract_VM_Version {
 //   z196: 2010-08
 //   ec12: 2012-09
 //   z13:  2015-03
+//   z14:  2017-09
+//   z15:  2019-09
```
Comment (TheRealMDoerr) [inline src/hotspot/cpu/s390/vm_version_s390.hpp:48]: How does this relate to the table in the .cpp file? I'd prefer to have such kind of information consolidated at one place.

**Review by TheRealMDoerr — APPROVED** [review]
Detection looks good to me.
Cleanup proposal: Try to make dates more consistent and comprehensive.

File: src/hotspot/cpu/s390/vm_version_s390.cpp (hunk 1 above)
Comment (RealLucy) [reply src/hotspot/cpu/s390/vm_version_s390.cpp:68]: You are right, as of now, the tables are mostly for documentation. z_EOS is used now to fill _features_string. I'd like to keep the tables as they are for possible future use.

File: src/hotspot/cpu/s390/vm_version_s390.cpp (hunk 2 above)
Comment (RealLucy) [reply src/hotspot/cpu/s390/vm_version_s390.cpp]: Changed such that End-of-Support information is taken from table z_EOS[].

**Review by TheRealMDoerr — APPROVED** [review]
Thanks for cleaning up! LGTM.

**Comment by RealLucy** [comment]
Goetz and Martin,
thanks a lot for your reviews. 
Lutz

**Comment by RealLucy** [comment]
Build errors on Win-x64 are unrelated to these s390-only changes.

**Pull request closed without merging**